src/main.rs

```diff
@@ -668,8 +694,9 @@ fn exec_cargo_inner(
     cx: &Context,
     id: &PackageId,
     line: &mut ProcessBuilder<'_>,
-    progress: &mut Progress,
+    progress: &Arc<Mutex<Progress>>,
 ) -> Result<()> {
+    let mut progress = progress.lock().unwrap();
```

Comment:
I believe this guard is held until the execution of the command is completed, so I'm concerned that this may cause the executions to not be parallel.

Reply:
I added some lexical scoping to limit how long these locks are held. This did involve moving the count increment because of the off chance that 2 threads would increment before either got to print their count. In general, I'm not sure how to look for more cases of this forced serialization.